nixos/modules/services/web-servers/webhook.nix

```diff
@@ -0,0 +1,80 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.webhook;
+  webhook = pkgs.webhook;
+  user = cfg.user;
+  settingsJSON = pkgs.writeText "webhookSettings" (builtins.toJSON cfg.settings);
+
+in {
+  options.services.webhook = {
+
+    enable = mkOption {
+      default = false;
+      description = "Enable webhook service.";
+      type = types.bool;
+    };
+
+    user = mkOption {
+      default = "webhook";
+      description = "User webhook service runs as.";
+      type = types.str;
+    };
+
+    group = mkOption {
+      default = null;
+      description = "Group webhook service runs as.";
+      type = with types; nullOr str;
+    };
+
+    settings = mkOption {
+      default = null;
+      description = "Webhook service settings.";
+      type = with types; listOf attrs;
```

Comment:
since this seems to be used only to generate json files, check out `pkgs.formats.json`